adapters/repos/db/lsmkv/memtable_flush.go

```diff
@@ -143,6 +143,8 @@ func (l *Memtable) flushDataSet(f io.Writer) ([]keyIndex, error) {
 }
 
 func (l *Memtable) flushDataMap(f io.Writer) ([]keyIndex, error) {
+	l.RLock()
+	defer l.RUnlock()
```

Comment:
Instead of deferring, could we unlock immediately after calling `flattenInOrder()`, I assume that would hold the lock for a much shorter duration since this method does a lot more? Or do we need the lock for the entire duration?

Reply:
right, locks just around `flattenInOrder` should be sufficient.
`values` used later on are copy of the original ones, so there is no risk of parallel access